oeps/oep-0051-bp-conventional-commits.rst

```diff
@@ -0,0 +1,167 @@
+
+The nearer to the change that information is captured, the easier it will be to write, and the more accurate it will be.  Changes made by developers are consumed by many people, from the pull request reviewer to the final user of the product.  The more the developer can explain the change, the more easily the change will be understood and used, and the more efficient the overall development process will be.
+
+Commit messages are the closest text to the change itself.  The more
+information-rich we can make commit messages, the more useful they will be.
+
+
+Specfication
+============
+
+We are adopting the `Conventional Commits`_ spec, with our own adjustments.
+
+.. note:: Many people are familiar with the `Angular commit message format`_, which uses conventional commits and inspired the Conventional Commits spec.  We are **not** adopting the Angular rules.
+
+Git commit messages have these parts::
+
+    <type>: <subject>
+
+    <body>
+
+    <footer>
+
+Type
+----
+
+We use these commit types:
+
+* **build**: anything to do with building or releasing the repo: Makefile, tox.ini, Travis, Jenkins, GitHub actions, and so on.
+
+* **chore**: a repetitive mechanical task such as updating requirements or translations.
+
+* **docs**: documentation-only changes. Documentation changes for other types of work should  be included in those commits.
+
+* **feat**: a new feature, or a change to an existing feature. Anything that changes the set of features.
+
+* **fix**: bug fixes, including changes to the behavior of features, and changes that remove or mitigate security vulnerabilities.
+
+* **perf**: performance improvement.
+
+* **refactor**: code reorganization that doesn't change behavior from the repo consumer perspective.
+
+* **revert**: undo a previous change. The previous commit message, including the type prefix, is included.
+
+* **style**: improve the styling of the code.
+
+* **test**: test-only changes. Adds missing tests or corrects existing tests. Tests accompanying other types of changes go into those commits.
+
+Breaking changes include an exclamation point as part of the type::
+
+    feat!: removed the ability to author courses
+
+Ideally, a single commit won't mix these types together.  If a commit does, use the most important type in the commit message.  The priority order for the types is generally: revert, feat, fix, perf, docs, test, build, refactor, style, chore.
+
+Scope
+-----
+
+The `Conventional Commits`_ spec includes a parenthesized scope after the type.  Open edX repos are large and varied, making standardization of scopes difficult.  We won't use scopes for now.  This could change in the future.
+
+Subject
+-------
+
+Commit message subjects should be short.  Put more information in the body of the commit message to fully explain your change.
```

Comment:
It's currently common at edX to put the ticket number at the beginning of the subject line; eg `JIRA-1234 we made a change`

In spirit of keeping commit subjects short, should we recommend that ticket numbers (if included) should go in the body?

Reply:
Or at the end of the subject so they still show up in `git log --oneline`, but don't get in the way as much

Reply:
Including them at the end of the subject seems good.